include/deal.II/grid/manifold_lib.h

```diff
@@ -127,6 +115,66 @@ class SphericalManifold : public ChartManifold<dim, spacedim, spacedim>
 
 
 /**
+ * Manifold description for a spherical space coordinate system.
+ *
+ * You can use this Manifold object to describe any sphere, circle,
+ * hypersphere or hyperdisc in two or three dimensions, both as a co-dimension
+ * one manifold descriptor or as co-dimension zero manifold descriptor.
+ *
+ * The two template arguments match the meaning of the two template arguments
+ * in Triangulation<dim, spacedim>, however this Manifold can be used to
+ * describe both thin and thick objects, and the behavior is identical when
+ * dim <= spacedim, i.e., the functionality of SphericalManifold<2,3> is
+ * identical to SphericalManifold<3,3>.
+ *
+ * @ingroup manifold
+ *
+ * @author Mauro Bardelloni, Luca Heltai, 2016
+ */
+template <int dim, int spacedim = dim>
+class SphericalManifold : public Manifold<dim, spacedim>
+{
+public:
+  /**
+   * The Constructor takes the center of the spherical coordinates system.
+   * This class uses the pull_back and push_forward mechanism to transform
+   * from Cartesian to spherical coordinate systems, taking into account the
+   * periodicity of base Manifold in two dimensions, while in three dimensions
+   * it takes the middle point, and project it along the radius using the
+   * average radius of the surrounding points.
+   */
+  SphericalManifold(const Point<spacedim> center = Point<spacedim>());
+
+  /**
+   * TODO:
+   */
+  virtual
+  Point<spacedim>
+  get_new_point(const Point<spacedim> &P1, const Point<spacedim> &P2, double w) const;
```

Comment:
For consistency with the rest of the library, please also use `p1`, `p2` (i.e., lower-case) and mark `w` as `const` (because it is an *input* argument).